index.html

```diff
@@ -545,14 +547,29 @@ <h4>The MediaStreamTrackAudioStats interface</h4>
               such as if the track is muted or disabled, then the counters do
               not increase.</p>
             </div>
+            <li>
+              <p><dfn data-lt="input latency">Input latency</dfn> is the time,
+              in milliseconds, between the point in time an audio input device
+              has acquired a signal and the time it is available for
+              consumption, which may include buffering by the user agent.</p>
+              <p>The <dfn data-lt="latest input latency">latest input
+              latency</dfn> is the latest available [= input latency =] as
+              estimated between the track's input device and delivery to any of
+              its sinks.</p>
+              <div class="note">
+                <p>A sink that consumes audio may add additional processing
+                latency not included in this measurement, such as playout delay
+                or encode time.</p>
```

Comment:
I do not think the intent of this value is for realtime processing.
Adding a note about the intent (goal is to allow monitoring not realtime processing) would be nice.
If we want very precise latency info from say web audio, it might be best to expose this value in web audio measuring microphone to web audio sink.

Reply:
I added a note about monitoring, WDYT @youennf ?

Reply:
If that value is useful, why not adding it to WebAudio directly? This should be cheap enough to do and would give a more precise latency info given it would compute the actual latency between microphone and web audio graph.

Reply:
The Web Audio API does not deal with input, only output. It's problematic for various reasons, but it's how it was done. The object that is used to refer to the microphone is the `MediaStreamTrack`, and so this is why we're adding it there.

The latency in between the microphone and the web audio graph is precisely what is being added here, in the sense that the web audio graph is a sink, because it runs off of a system audio output callback.

But there can be other sinks, such as an `HTMLMediaElement`, or using the microphone in some other way, directly feeding it to a `MediaRecorder`, etc.

Reply:
Since this is just a note, if the rest of the PR looks good, can we merge and iterate on the design? I'd like to do a PR for the min/max/avg latencies as a follow up

Reply:
> The Web Audio API does not deal with input,
> The object that is used to refer to the microphone is the `MediaStreamTrack`

I do not understand this.
`MediaStreamAudioSourceNode` is the web audio sink of `MediaStreamTrack`.
Can you clarify why microphone-to-web-audio latency should not be put there instead?
At least in Safari, there is a small buffer at `MediaStreamAudioSourceNode` which could impact the web audio input latency.

> there can be other sinks, such as an `HTMLMediaElement`

Exactly, with different latencies potentially. It makes more sense to put values there for those applications.
For instance, at least for Safari's implementation, droppedVideoFrames makes sense in https://w3c.github.io/media-playback-quality/#dom-htmlvideoelement-getvideoplaybackquality but does not make sense in`MediaStreamTrack`.

Reply:
Re dropped video frames, in Chrome it is possible for frames to be dropped prior to sink delivery (measured by `track.stats`) or by the sink, for example the RTCPeerConnection (measured by `pc.getStats`). It would clearly be wrong to expose encoder frame drops inside `track.stats`. We could expose track frame drops inside the `media-source` stats object that is returned by `getStats`, but this would mean that every sink has to report something that is outside of its own scope. And let me remind you that the reason we're talking about `track.stats` rather than `getStats` is that when we added these and other metrics to webrtc-stats, we got pushback for adding stuff that is out of scope.

So likewise, if `MediaStreamAudioSourceNode` is adding latency that is not accounted for by `track.stats`, then that seems like a good place to measure it. But I still see value in each part of the pipeline measuring its own thing rather than letting components be responsible for neighbor components and making assumptions about how many steps in the pipeline is of interest to the application

Reply:
I agree with @henbos. If `MediaStreamAudioSourceNode` (or its `Track` sibling) adds latency, then we can add something there. More likely than not we won't add something there, because if it adds latency there is a design problem in the implementation, and we'll argue in the exact same way we're arguing about audio frame drops: it's better for the Web that we fix our implementation than to add a way to measure avoidable latency overhead.

Reply:
If we agree, does that mean we can merge this PR?